OMEdit/OMEditLIB/OMS/ElementPropertiesDialog.cpp

```diff
@@ -403,3 +405,89 @@ void ElementPropertiesDialog::updateProperties()
   // accept the dialog
   accept();
 }
+
+/*
+ * event filter for mParameterLineEdits and mInputLineEdit
+ * to detect the focus out event, and update
+ * the default start values from modeldesctiption.xml
+ */
+bool ElementPropertiesDialog::eventFilter(QObject *pObject, QEvent *pEvent)
+{
+  QLineEdit *pLineEdit = qobject_cast<QLineEdit*>(pObject);
+
+  if (pLineEdit && pEvent->type() == QEvent::FocusOut) {
+    if (pLineEdit->text().isEmpty()) {
+      // search the lineEdit index in parameters or inputs
+      int parameterIndex = mParameterLineEdits.indexOf(pLineEdit);
+      int inputIndex = mInputLineEdits.indexOf(pLineEdit);
+
+      if (parameterIndex != -1) {
+        QString pParameterLabelText = mParameterLabels.at(parameterIndex)->text();
+        updateStartValues(pParameterLabelText, pLineEdit);
+      }
+
+      if (inputIndex != -1) {
+        QString pInputLabelText = mInputLabels.at(inputIndex)->text();
+        updateStartValues(pInputLabelText, pLineEdit);
+      }
+    }
+  }
+  return QWidget::eventFilter(pObject, pEvent);
+}
+
+/*
+ * helper function to updateStartValues read from modeldescription.xml
+ */
+void ElementPropertiesDialog::updateStartValues(QString pName, QLineEdit * pLineEdit)
+{
+  if (mpComponent->getLibraryTreeItem()->getOMSElement() && mpComponent->getLibraryTreeItem()->getOMSElement()->connectors) {
+    oms_connector_t** pInterfaces = mpComponent->getLibraryTreeItem()->getOMSElement()->connectors;
+    for (int i = 0 ; pInterfaces[i] ; i++) {
+      QString nameStructure = QString("%1.%2").arg(mpComponent->getLibraryTreeItem()->getNameStructure(), QString(pInterfaces[i]->name));
+      if (pInterfaces[i]->causality == oms_causality_parameter && QString(pInterfaces[i]->name)== pName) {
+        OMSProxy::instance()->omsDelete(nameStructure + ":start");
+        updateStartValuesHelper(pInterfaces[i], nameStructure, pLineEdit);
+        break;
```

Comment:
Why is there a break statement?

Reply:
when we found the correct connector name to be updated,  we should break the loop, The connectors are read from the library tree browser and we need to iterate them to find the connector name and its type to be updated

Reply:
Why don't you just `return` then?

Reply:
yes we can do that

Reply:
this is fixed with https://github.com/OpenModelica/OpenModelica/pull/6857/commits/f59f561b36c5f56c9cd1279a50719ebe06a95fd3